kotlin-inject-compiler-core/src/main/kotlin/me/tatarka/inject/compiler/TypeCollector.kt

```diff
@@ -247,6 +247,9 @@ class TypeCollector private constructor(private val provider: AstProvider, priva
                 scopedComponent = scopedComponent?.type
             )
         }
+        if (astClass.isObject) {
```

Comment:
I'm thinking what if we move this to be above the `findInjectConstructors` call and remove the check inside the `findInjectConstructors`?

Reply:
This does work but it then findInjectConstructors returns a result for object classes which is incorrect since it generates code which doesn't compile.

Reply:
Remove check from here and added it to `ModelAstClass#primaryConstructor` to prevent it from incorrectly returning constructors for object classes.